modules/5-elixir.livemd

```diff
@@ -204,4 +205,47 @@ user_input = "some_string_which_obviously_isnt_the_same_as_the_password"
 # if SecurityCheck.validate(user_input, password) || raise(SecurityCheck) do :you_let_a_baddie_in end
 ```
 
+## Protecting Sensitive Data
+
+### Description
+
+Sensitive data is any information that should be out of reach from all outsiders unless they have permission to access it, which in most cases would be considered "confidential data". Some examples of sensitive data are PHI (Protected Health Information) or PII (Personally Identifiable Information).
+
+### Prevention
+
+I think we can all agree that making sure this information is secure is important, but let's dive into how to actually protect it:
+
+## Wrapping
+
+Exceptions may result in console or log output that includes a stack trace. Mostly a stack trace shows the module/function/arity and the filename/line where the exception occurred, but for the function at the top of the stack the actual list of arguments may be included instead of the function arity.
+To prevent sensitive data from leaking in a stack trace, the value may be wrapped in a closure: a zero-arity anonymous function. The inner value can be easily unwrapped where it is needed by invoking the function. If an error occurs and function arguments are written to the console or a log, it is shown as #Fun<...> or #Function<...>. Secrets wrapped in a closure are also safe from introspection using Observer and from being written to crash dumps.
+
+### <span style="color:blue;">Example</span>
+
+```elixir
+wrapped_secret = fn -> System.get_env("SECRET") end
+```
+
+## Stacktrace Pruning
+
+Another approach, useful in functions that call the standard library (e.g. crypto) or other functions that do not support wrapping secrets in a closure, is stripping argument values from the stack trace when an exception occurs. This can be done by wrapping the function call(s) in a try … catch expression (Erlang) or adding a rescue clause to a function body (Elixir), and stripping the function arguments before re-raising the exception:
+
+## ETS Tables
+
+ETS tables can be declared as ‘private’ (:private option), preventing the table from being read by other processes, such as remote shell sessions. Private tables are also not visible in ‘observer’.
+
+### <span style="color:red;">Quiz</span>
+
+**We have decided that we do not want this ETS table to be read from other processes, so try making it private:**
```

Comment:
This is a great quiz question! In its current form however, it won't provide much feedback that is actionable for the grader. I think one way to improve this question would be to setup a module/function that attempts to read from the cool_table that works only when the `:private` option **isn't** set and then errors out when it can no longer read from it (meaning the quiz taker has submitted the correct answer)

Reply:
wont the output be very different as is:
If it is public it would show the empty table, if it is private it would show #Reference<0.3157030696.1001783298.195388> or some other random number 

Should be fairly easy to compare those outputs

Reply:
Current output in iex when you run `cool_table = :ets.new(:cool_table, [])` is you get the binary output `#Reference<...>` and when you run the secure correction (`cool_table = :ets.new(:cool_table, [:private])`) you'll get the same output (with just a different address in memory).

That memory object is going to be sent across the wire as a string, which we can't compare to an answer key.

Reply:
ahhh gotcha, Ill make the change should be an easy fix